File: javascript/problem035.js
var array = new Array(1000000);

for(i=2; i<Math.sqrt(1000000); i++){
  if(array[i] == undefined){
    for(j=i+i; j<1000000; j = j + i){
      array[j] = false;
    }
  }
}

function isCircularPrime(n){
  var str = n.toString();
  for(a=0; a<str.length - 1; a++){
    var tempStr = str.substring(a+1, str.length) + str.substring(0, a+1);
    var tempValue = parseInt(tempStr);
    if(array[tempValue] != undefined){
      return false;
    }
  }
  return true;
}

var result = 0;
for(i=2; i<1000000; i++){
  if(array[i] == undefined && isCircularPrime(i)){
    result = result + 1;
  }
}

console.log(result);

File: python/problem035.py
import math

number_array = [True] * 1000000

for i in range(2, int(math.sqrt(1000000))):
    if number_array[i]:
        j = i + i
        while j < 1000000:
            number_array[j] = False
            j += i


def is_circular_prime(n):
    str_n = str(n)
    for a in range(0, len(str_n) - 1):
        temp_str = str_n[a + 1:len(str_n)] + str_n[0:a + 1]
        temp_value = int(temp_str)
        if not number_array[temp_value]:
            return False

    return True


result = 0
for i in range(2, 1000000):
    if number_array[i] and is_circular_prime(i):
        result += 1
print(result)
